shared/js/background/utils.es6.js

```diff
@@ -114,6 +140,45 @@ function getAsyncBlockingSupport () {
     return false
 }
 
+/*
+ * check to see if this is a broken site reported on github
+*/
+function isBroken (url) {
+    if (!tdsStorage || !tdsStorage.brokenSiteList) return
+
+    const parsedDomain = tldts.parse(url)
+    let hostname = parsedDomain.hostname || url
+
+    // If root domain in temp whitelist, return true
+    return tdsStorage.brokenSiteList.some((brokenSiteDomain) => {
+        if (brokenSiteDomain) {
+            return hostname.match(new RegExp(brokenSiteDomain + '$'))
+        }
+    })
+}
+
+// return true if the given url is in the safelist. For checking if the current tab is in the safelist,
+// tabManager.site.whitelisted is the preferred method.
+function isSafeListed (url) {
+    const hostname = extractHostFromURL(url)
+    const safeList = settings.getSetting('whitelisted')
+    let subdomains = hostname.split('.')
+    // Check user safe list
+    while (subdomains.length > 1) {
+        if (safeList && safeList[subdomains.join('.')]) {
+            return true
+        }
+        subdomains.shift()
+    }
+
+    // Check broken sites
+    if (isBroken(hostname)) {
```

Comment:
it looks like `isBroken` takes full url and not a hostname. Not sure if that matters.

Reply:
That's fine - it only needs the hostname, and if it gets a full URL it drops the rest, so it should be fine here. Thanks for everything!